internal/dkg/store.go

```diff
@@ -185,30 +124,49 @@ func (s *BoltStore) MigrateFromGroupfile(beaconID string, groupFile *key.Group,
 	if share == nil {
 		return errors.New("you cannot migrate without a previous distributed key share")
 	}
-	// we use a separate lock here to avoid reentrancy when calling `.SaveFinished()`
-	s.migrationLock.Lock()
-	defer s.migrationLock.Unlock()
```

Comment:
why is the reentrancy no longer a risk?

Reply:
that being said, SaveFinished didn't have a lock, so I'm unclear what reentrancy issue we had, not a lock contention in any case.

Reply:
At some point we were providing a CLI command to do the migration, so perhaps this is a carryover from then

Reply:
ah yeah this is not 100% intentional. I mildly refactored this code because I needed to re-use some of the contained logic. That included factoring out the mutex, which we decided wasn't needed for the filestore because it was partitioned by beacon, and each beacon would have at most one process.

CLI command is a good point. I could reintroduce it for that reason.

Reply:
the CLI command doesn't exist anymore; i guess in theory it's possible for a really slow migration to overwrite a very fast initial DKG in some test case? though I don't think so

Reply:
woah unless this could be the reason for failing tests regarding group file migration :p don't think so though